x-pack/plugins/ingest_manager/common/services/package_to_config.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PackageInfo, RegistryDatasource, Datasource, DatasourceInput } from '../types';
+
+/*
+ * This service creates a datasource inputs definition from defaults provided in package info
+ */
+export const packageToConfigDatasourceInputs = (packageInfo: PackageInfo): Datasource['inputs'] => {
```

Comment:
@paul-tavares This is the service I mentioned that will help with #59564. It takes package info and creates the `inputs` definition for a datasource using defaults from the package info. [Here is an example](https://github.com/elastic/kibana/pull/59590/files#diff-340add3e41d1f6fb354ad2f2c693fe65R62) of where it is used.

Reply:
Thanks @jen-huang . I will work on it once this merges.